docs/user-guide/a1-06-spirv-target-specifics.md

```diff
@@ -0,0 +1,316 @@
+
+### [vk::restrict_pointer]
+Opposite to [vk::aliased_pointer] in a way that it tells the compiler to treat the variable as not-aliased.
+
+### [vk::spirv_instruction(op : int, set : String = "")]
+When applied to a function, the function will use the `op` value corresponding to a SPIR-V instruction. You can also specify which instruction set you want to use as a second argument. E.G., `[[vk::spirv_instruction(1, "NonSemantic.DebugBreak")]]`
+
+
+Multiple entry points support
+-----------------------------
+
+Slang supports mutiple entry points when targeting SPIR-V.
+It is same to HLSL and SPIR-V that allows a shader source to have multiple entry points.
+Note that GLSL requires the entry point to be named, "main", and a shader source can have only one entry point.
+
+
+Memory pointer is experimental
+------------------------------
+
+SPIR-V currently supports five types of `Addressing model`-s, and it can be grouped as Logical addressing model and non-local addressing model.
+
+The logical addressing model means that the pointers are abstract, and they have no physical size nor numeric value.
+
+The non-Logical addressing models allow physical pointers to be formed. When a pointer is declared, it is declared internally with a specific `Storage Class`.  SPIR-V has a limitation that "Pointers for one Storage Class must not be used to access objects in another Storage Class."
+
+Slang supports the Logical and non-Logical addressing models with the limitations from SPIR-V and addional limitations. See more [explanation and an example](convenience-features.html#pointers-limited).
+
+Some examples of the `Storage class` are listed below:
+- UniformConstant: is for uniform constant variables as read-only.
+- Uniform: is for uniform buffer as read-only.
+- StorageBuffer: is for UAV/structuredBuffer/ShaderStorageBuffer as read-write.
+- PhysicalStorageBuffer: same as StorageBuffer but it follows non-logical addressing model.
+- PushConstant: is for variables as read-only stored in the buffer that has `push_constant` layout.
+
+Note that only ones starting with `Physical-` follows the non-logical addressing model.
+
+
+Difference between `std140` layout and `std430` layout
+------------------------------------------------------
+
+When constant parameters are send to the shader, there are different rules applied to the alignment and paddings.
+`std140` requires alignment of 16 bytes on `struct` types and it adds padding to round up to 16 bytes for each member.
+`std430` requires a natural alignment, which means that the required alignment is same as its size.
+
+
+Matrix type translation
+-----------------------
+
+When targeting SPIR-V, Slang assumes Row-Major matrix, which is same behavior as HLSL/DXC.
```

Comment:
I think  this is more a terminology difference rather than an actual divergence.

Matrix type translation is orthogonal to row-major/column-major layout.

In this section we should focus on how matrix types are translated, and not touch into row-major and column-major.

Just have a table showing how hlsl matrices are translated to spirv types, similar to that table in the dxc doc. And say that we follow the same behavior in dxc.

Then we can go for the reasoning behind this design, in that what hlsl calls a row is what glsl will call a column. Essentially, a hlsl row or glsl column is:
- what `matrix[i]` returns.
- what matrices can be constructed from `float3x4(v1, v2, v3)`, or `mat3x4(v1,v2,v3)`.
If we map it the other way, simple operations like `matrix[i]` or construction will become very messy code that may lead to slower performance.

The only consequence of mapping float3x4 to `matrix(vector4, 3)` (or mat3x4, a 3-"column"-by-4-"row" matrix in spirv terminology) is that matrix-vector and matrix-matrix multiplication operations needs to have their operand ordering swapped, which is what slang will do:
```
mul(m, v) ==> v*m
```

Reply:
Once we talk about this, then there is the row-major and column-major layout. Note that the decision to use row-major or column-major layout has nothing to do with how the matrix type itself is defined in hlsl or spirv. It really just a data layout modifier just like `std140` or `std430`.

We can talk about this in a separate section saying that by default, slang uses row-major layout for reasons documented in [a1-01-matrix-layout], as oppposed to defaulting to column major layout in dxc.

Reply:
I will change it as suggested.